tensorflow_io/ignite/kernels/ggfs/ggfs_client.cc

```diff
@@ -267,10 +266,9 @@ Status GGFSClient::ReceiveCommonResponseHeader() {
       int32_t err_msg_length;
       TF_RETURN_IF_ERROR(client_->ReadInt(&err_msg_length));
 
-      uint8_t *err_msg_c = new uint8_t[err_msg_length];
-      auto clean = gtl::MakeCleanup([err_msg_c] { delete[] err_msg_c; });
-      TF_RETURN_IF_ERROR(client_->ReadData(err_msg_c, err_msg_length));
-      string err_msg(reinterpret_cast<char *>(err_msg_c), err_msg_length);
+      std::unique_ptr<uint8_t> err_msg_c = std::unique_ptr<uint8_t>(new uint8_t[err_msg_length]);
```

Comment:
@dmitrievanthony wondering if this should be `std::unique_ptr<uint8_t> err_msg_c(new uint8_t[err_msg_length])`, or a customized deleter?

I know some compilers will automatically call `delete []` even if `delete` is called for array, e.g., `uint_8 arr=new uint8[10]; delete arr;`. However, we are still using Ubuntu 14.04's default GCC (4.8) so it may or may not be able to call `delete []` automatically. Think we could explicitly calling the deleter for array?